runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/QueryParameters.kt

```diff
@@ -33,13 +31,12 @@ private object EmptyQueryParameters : QueryParameters {
     override fun isEmpty(): Boolean = true
 }
 
-class QueryParametersBuilder : StringValuesMapBuilder(true, 8) {
+class QueryParametersBuilder : StringValuesMapBuilder(true, 8), CanDeepCopy<QueryParametersBuilder> {
     override fun toString(): String = "QueryParametersBuilder ${entries()} "
-    override fun build(): QueryParameters {
-        require(!built) { "QueryParametersBuilder can only build a single instance" }
-        built = true
-        return QueryParametersImpl(values)
-    }
+    override fun build(): QueryParameters = QueryParametersImpl(values)
+
+    override fun deepCopy(): QueryParametersBuilder =
+        QueryParametersBuilder().apply { values.putAll(this@QueryParametersBuilder.values.deepCopy()) }
```

Comment:
style

I found this hard to read/follow due to `this` both referring to the same type `QueryParametersBuilder`. I guess it's fine as is if it works but it took me staring at this for a second to verify it was correct.